src/lib/node.ts

```diff
@@ -209,3 +209,88 @@ export function pipeToNodeStream<T>(
 		handleTsStreamError // abort handler
 	);
 }
+
+type FromNodeReadableDataType<T> = T extends void ? string | Buffer : T;
+/**
+ * Create a ts-stream ReadableStream from a Node.JS ReadableStream.
+ *
+ * Reads all values from `nodeReadable` and writes them to the returned stream.
+ * When the `nodeReadable` ends, the returned stream is also ended.
+ * Should the `nodeReadable` emit a `close` event, the returned stream is ended without error.
+ * If the `nodeReadable` emits a `error` event, the returned stream is ended with the given error.
+ *
+ * Usage example:
+ * ```ts
+ * const source = fromNodeReadable(fs.createReadStream('source.txt'));
+ * source
+ *  .map((chunk) => String(chunk))
+ *  .map((letters) => letters.toUpperCase())
+ *  .forEach(console.log);
+ * ```
+ *
+ * @param nodeReadable Soource NodeJS stream
+ * @returns Readable stream of type `string | Buffer`
+ */
+export function fromNodeReadable<T = void>(
+	nodeReadable: NodeJS.ReadableStream
+): ReadableStream<FromNodeReadableDataType<T>> {
+	const stream = new Stream<FromNodeReadableDataType<T>>();
+
+	const endStream = async (err?: Error) => {
+		if (!stream.isEndingOrEnded()) {
+			/**
+			 * `stream.end` could reject, so we'll emit
+			 * any caught errors as an event via the NodeJS stream.
+			 * This is acceptable as we are only attempting to end
+			 * the ts-stream a maximum one time.
+			 */
+			try {
+				return stream.end(err);
+			} catch (e) {
+				nodeReadable.emit("error", e);
+			}
+		}
+	};
+
+	stream.aborted().catch(endStream);
+	nodeReadable.once("close", endStream);
+	nodeReadable.once("error", endStream);
+
+	(async () => {
+		try {
+			for await (const chunk of nodeReadable) {
+				await stream.write(chunk as FromNodeReadableDataType<T>);
+			}
+			await endStream();
+		} catch (err) {
+			await endStream(err as Error);
```

Comment:
Sorry, forgot to comment on this one in the first review...
It is better to ensure `err` really is an Error. Something like:

```ts
await endStream(err instanceof Error ? err : new Error(`${err}`));
```

Not the prettiest, but at least it guarantees it's an Error, and will work for strings (which I suppose is the most common non-Error thing someone would 'throw').

Reply:
Added a helper function that's now being used in a few places to ensure we actually have an instance of an Error.